### PR TITLE
Htcondor 1658 general improvements branch

### PR DESCRIPTION
**Pull request by ColeBollig**

### CODE REVIEW

File: src/HTCondor/condor-9-to-10-upgrade-check.py [hunk 1]
```diff
@@ -327,40 +365,33 @@ def check_pcre2():
...
-                8,
-                err=True,
-            )
-            return
+        format_print(
+            """Failed to find 'pcre2grep' command. Will do a simple check for incompatibilties.
+        Recommended to install 'pcre2grep' command for better check of map files.""",
+            offset=8,
+            err=True,
+        )
```
Comment (jasoncpatton) [inline src/HTCondor/condor-9-to-10-upgrade-check.py]: I didn't even think about what `offset` would do with multiline messages, but testing shows this works right. I might suggest omitting it for multiline messages, though, and put whatever spaces are needed in the string itself:

```suggestion
        format_print(
            """        Failed to find 'pcre2grep' command. Will do a simple check for incompatibilties.
        Recommended to install 'pcre2grep' command for better check of map files.""",
            err=True,
        )
```

File: src/HTCondor/condor-9-to-10-upgrade-check.py [hunk 2]
```diff
@@ -404,43 +434,62 @@ def check_gpu_requirements():
...
 def main():
-    # Make sure script is running as root because we need to run 'condor_token_list'
-    if os.geteuid() != 0:
-        format_print("Error: Script not ran as root")
-        sys.exit(1)
     # Systems with config to check: Base condor and condor-ce
     ecosystems = ["HTCondor"]
+    check_gpu_reqs = True
     # Check if passed -ce flag to indicate this host is a CE
     for arg in sys.argv:
```
Comment (jasoncpatton) [inline src/HTCondor/condor-9-to-10-upgrade-check.py]: Ah, I forgot to mention that we should probably be using the `argparse` library here, but it's probably a bit too late for that. Anyway, the first arg is always the program name so can probably skip that:

```suggestion
    for arg in sys.argv[1:]:
```

**Review by jasoncpatton — CHANGES_REQUESTED** [review]
It would be nice to run this through `black` again, if it's easy for you to `pip` or `conda` install it quickly in a virtual environment. Otherwise, no worries.

**Comment by ColeBollig** [comment]
@jasoncpatton I merged your suggestions and ran black on the script again

**Review by jasoncpatton — APPROVED** [review]

**Comment by jasoncpatton** [comment]
> @jasoncpatton I merged your suggestions and ran black on the script again

LGTM!

**Pull request merged**